common/src/main/java/com/microsoft/identity/common/internal/controllers/ApiDispatcher.java

```diff
@@ -405,12 +413,18 @@ public void run() {
                     handler.post(new Runnable() {
                         @Override
                         public void run() {
+                            // end api event
+                            Telemetry.emit(
+                                    new ApiEndEvent()
```

Comment:
What's the reason we're putting  ApiEndEvent() here?

Asking because it seems very 'out of place', as in I'm not seeing any other ApiEndEvent() and ApiStartEvent() in other functions. Are they being handled in other layers? If that's the case, shall we move this event up there (or bring all of them down here)?

Reply:
Start and End events do not need to be in the same layer. This is the flexibility of the design here. It emits a start event before sending a request and emits an end event when either getting a result or catching an exception, like here. As the exception could be thrown anywhere in the nested calls and I do not want to add try-catch and emit it everywhere. This is the right place to emit the api end event when the api call ends with any exception throwing in the middle.

Reply:
Agree the Telemetry design gives you the flexibility . But why is there only one emit event here on exception ? Ideally I would see API end events for success case as well here in APIDispatcher (both acquire token and acquire token silent). Any reason why we are not doing it?

Reply:
We can delegate this discussion to later as well considering we want to merge this soon, since this is about where to emit the events.

Reply:
Maybe some confusion here. The plug-in here is not done yet. This was for testing purpose before.